components/style/values/specified/mod.rs

```diff
@@ -339,47 +367,61 @@ impl ToComputedValue for Angle {
     }
 
     fn from_computed_value(computed: &Self::ComputedValue) -> Self {
-        Angle {
-            radians: computed.radians(),
-            was_calc: false,
-        }
+        Angle::from_radians(computed.radians())
     }
 }
 
 impl Angle {
+    /// Returns an angle with the given value in degrees.
+    pub fn from_degrees(value: CSSFloat) -> Self {
+        Angle { value: value, unit: AngleUnit::Degree, was_calc: false }
+    }
+    /// Returns an angle with the given value in gradians.
+    pub fn from_gradians(value: CSSFloat) -> Self {
+        Angle { value: value, unit: AngleUnit::Gradian, was_calc: false }
+    }
+    /// Returns an angle with the given value in turns.
+    pub fn from_turns(value: CSSFloat) -> Self {
+        Angle { value: value, unit: AngleUnit::Turn, was_calc: false }
+    }
+    /// Returns an angle with the given value in radians.
+    pub fn from_radians(value: CSSFloat) -> Self {
+        Angle { value: value, unit: AngleUnit::Radian, was_calc: false }
+    }
+
     #[inline]
     #[allow(missing_docs)]
     pub fn radians(self) -> f32 {
-        self.radians
-    }
+        use self::AngleUnit::*;
 
-    /// Returns an angle value that represents zero radians.
-    pub fn zero() -> Self {
-        Self::from_radians(0.0)
+        const RAD_PER_DEG: CSSFloat = PI / 180.0;
+        const RAD_PER_GRAD: CSSFloat = PI / 200.0;
+        const RAD_PER_TURN: CSSFloat = PI * 2.0;
+
+        let radians = match self.unit {
+            Degree => self.value * RAD_PER_DEG,
+            Gradian => self.value * RAD_PER_GRAD,
+            Turn => self.value * RAD_PER_TURN,
+            Radian => self.value,
+        };
+        radians.min(f32::MAX).max(f32::MIN)
     }
 
-    #[inline]
-    #[allow(missing_docs)]
-    pub fn from_radians(r: f32) -> Self {
-        Angle {
-            radians: r.min(f32::MAX).max(f32::MIN),
-            was_calc: false,
-        }
+    /// Returns an angle value that represents zero.
+    pub fn zero() -> Self {
+        Self::from_degrees(0.0)
     }
 
     /// Returns an `Angle` parsed from a `calc()` expression.
     pub fn from_calc(radians: CSSFloat) -> Self {
         Angle {
-            radians: radians.min(f32::MAX).max(f32::MIN),
+            value: radians.min(f32::MAX).max(f32::MIN),
```

Comment:
Per the other conversation, this isn't really needed I guess, but I don't think it hurts, feel free to fix it up if you have the time before bors and treeherder report results.